File: backend/components/metiers/threed/tests_centralises/api/accessibility/test_accessibility.py
from threed.api.accessibility.accessibility import check_accessibility


class TestAccessibility:
    def test_accessibility_basic(self):
        entity = {"label": "Test"}
        assert check_accessibility(entity) is True
        assert check_accessibility({}) is False
        assert check_accessibility(None) is False

    def test_accessibility_advanced(self):
        entity = {"alt": "Alternative"}
        assert check_accessibility(entity) is True
        # Cas limites
        assert check_accessibility({"foo": "bar"}) is False
        assert check_accessibility({"label": ""}) is True

File: backend/components/metiers/threed/tests_centralises/api/accessibility/test_accessibility.js
const mockFunctions = require("../../../mocks/functions_mock");
// test_accessibility.js – Test ultra avancé d'accessibilité API Threed (centralisé)

const { checkAccessibility } = require('../../../api/accessibility/accessibility');

describe('Accessibilité API Threed', () => {
  it('doit valider l’accessibilité d’une ressource complète', () => {
    const data = { label: 'Ultra', lang: 'fr' };
    const result = checkAccessibility(data);
    expect(result).toBe(true);
  });

  it('doit détecter une ressource non accessible (label manquant)', () => {
    const data = { lang: 'fr' };
    const result = checkAccessibility(data);
    expect(result).toBe(false);
  });
});
